datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java

```diff
@@ -122,6 +123,20 @@ public class EventProducer implements DatastreamEventProducer {
    * @param customCheckpointing decides whether producer should use custom checkpointing or the datastream server
    *                            provided checkpointing.
    */
+  public EventProducer(DatastreamTask task, TransportProvider transportProvider, CheckpointProvider checkpointProvider,
+      Properties config, boolean customCheckpointing) {
+    this(task, transportProvider, checkpointProvider, config, customCheckpointing, (t) -> new HashSet<>());
```

Comment:
Nit: Consider using `Collections.emptySet()` here when initializing your empty set. This is more intentional because it creates an immutable empty set.